pkg/configstore/store.go

```diff
@@ -0,0 +1,172 @@
+package configstore
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+// Store holds config data
+type Store struct {
+	// If you care about sanity, never write here, just read
+	Data  map[string]interface{}
+	Dir   string
+	Hints map[string]string
+	sync.Mutex
+
+	ListFiles  func(string) ([]string, error)
+	ParseFile  func(string, interface{}) error
+	FileExists func(string) (bool, error)
```

Comment:
TBW I'd rather have these as regular `func` and if we need some OOP then also an interface type. This "function as a member" does not seem very Golang-ish to me.

Reply:
my goal was to provide a config object with simple interface, which is golang-ish way (i think?), but i find golang-ish way to mock such objects in tests using interfaces very tedious compared to this approach